### PR TITLE
NN-5100 Get OIC Hearing results endpoint

### PR DESCRIPTION
**Pull request by Mjwillis**
Allows adj api to check if an hearing results has been added in NOMIS - in this instance will then lock down the report in the DPS service as its basically clobbered.

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/api/resource/AdjudicationsResource.java [hunk 1]
```diff
@@ -78,7 +78,7 @@ public ResponseEntity<AdjudicationCreationRequestData> requestAdjudicationCreati
     @ProxyUser
     @PreAuthorize("hasRole('MAINTAIN_ADJUDICATIONS') and hasAuthority('SCOPE_write')")
     public ResponseEntity<AdjudicationDetail> createAdjudication(@Valid @RequestBody @Parameter(description = "Adjudication details to save", required = true) final NewAdjudication adjudicationDetails) {
-        final var savedAdjudication = adjudicationsService.createAdjudication(adjudicationDetails.getOffenderNo(), adjudicationDetails);
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/api/resource/AdjudicationsResource.java:81]: Offender no not used in service

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 2]
```diff
@@ -156,8 +148,7 @@ public AdjudicationCreationRequestData generateAdjudicationCreationData(@NotNull
 
     @Transactional
     @VerifyOffenderAccess
-    public AdjudicationDetail createAdjudication(@NotNull final String offenderNo,
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:159]: Removed as not used

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 3]
```diff
@@ -83,11 +82,8 @@ public class AdjudicationsService {
     private final AgencyLocationRepository agencyLocationRepository;
     private final AdjudicationsPartyService adjudicationsPartyService;
     private final AgencyInternalLocationRepository internalLocationRepository;
-    private final AuthenticationFacade authenticationFacade;
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:86]: Some warning tidy ups

Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:86]: Some warning tidy ups

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 4]
```diff
@@ -259,7 +250,7 @@ public AdjudicationDetail updateAdjudication(@NotNull Long adjudicationNumber, @
             adjudication.getConnectedOffenderIds()
         );
 
-        final var updatedAdjudication = adjudicationsRepository.findByParties_AdjudicationNumber(adjudicationNumber).get();
+        final var updatedAdjudication = adjudicationsRepository.findByParties_AdjudicationNumber(adjudicationNumber).orElseThrow(EntityNotFoundException.withId(adjudicationNumber));
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:253]: Just warnings tidy up

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 5]
```diff
@@ -461,7 +470,11 @@ public List<Sanction> createOicSanctions(
 
         Long nextSanctionSeq = oicSanctionRepository.getNextSanctionSeq(result.getOffenderBookId());
 
-        List<OicSanction> oicSanctions = new ArrayList<>();
+        return transform(adjudicationNumber, oicSanctionRequests, result, nextSanctionSeq);
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:473]: Duplicate code tidy

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 6]
```diff
@@ -156,7 +148,7 @@ public AdjudicationCreationRequestData generateAdjudicationCreationData(@NotNull
 
     @Transactional
     @VerifyOffenderAccess
-    public AdjudicationDetail createAdjudication(@NotNull final String offenderNo,
+    public AdjudicationDetail createAdjudication(@SuppressWarnings("unused") @NotNull final String offenderNo, // This is to make the `@VerifyOffenderAccess` check access rights
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:151]: Add as was confusing why it was there.

File: src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java [hunk 7]
```diff
@@ -333,6 +325,24 @@ public void deleteOicHearing(final Long adjudicationNumber, final long oicHearin
         oicHearingRepository.delete(hearingToDelete);
     }
 
+    @Transactional(readOnly = true)
+    public List<OicHearingResultDto> getOicHearingResults(
```
Comment (Mjwillis) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AdjudicationsService.java:329]: Thought I'd return the list of results and Adj API can simple check for a result.

**Review by timoj58 — APPROVED** [review]

**Pull request merged**